python/tk_multi_loader/search_widget.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+from .ui.search_widget import Ui_SearchWidget
+from .utils import ResizeEventFilter
+
+class SearchWidget(QtGui.QWidget):
+    """
+    Small search box widget, similar to the one appearing when you click
+    CMD+f in chrome. This widget is typically parented with a QView of some 
+    sort, and when enable() is called, it will appear, overlayed on top of the
+    parent widget, in the top right corner. It has a text field where a search
+    input can be entered.
+    
+    You can connect to the filter_changed signal to get notified whenever the search
+    string is changed.
+    """
+    
+    # signal emitted whenever the search filter changes
+    filter_changed = QtCore.Signal(str)
+    
+    def __init__(self, parent):
+        """
+        Constructor
+        
+        :param parent: QT parent object
+        """
+        QtGui.QWidget.__init__(self, parent)
+
+        # make invisible by default
+        self.setVisible(False)
+        
+        # set up the UI
+        self._ui = Ui_SearchWidget() 
+        self._ui.setupUi(self)
+        
+        # now grab the default background color and use that
+        # in addition to that, apply the same styling that the search
+        # bar in the tree view is using. 
+        p = QtGui.QPalette()
+        bg_col = p.color(QtGui.QPalette.Active, QtGui.QPalette.Window)
+        
+        style = """
```

Comment:
it needs to pick up the default background from the style palette, in order to work nicely in for example both nuke and maya. This is why the style is built on the fly like this... I could have put evertything except the bg color into designer, but it seemed cleaner to keep everything in the same place!